src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -218,13 +218,19 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpControllerContext con
         [AllowAnonymous]
         public async Task<HttpResponseMessage> ExtensionWebHookHandler(string name, CancellationToken token)
         {
-            var provider = this._scriptHostManager.BindingWebHookProvider;
+            var provider = _scriptHostManager.BindingWebHookProvider;
 
             var handler = provider.GetHandlerOrNull(name);
             if (handler != null)
             {
-                var response = await handler.ConvertAsync(this.Request, token);
-                return response;
+                // key names for extension webhooks are named by convention
+                string keyName = $"{name}_extension".ToLowerInvariant();
+                if (!this.Request.HasAuthorizationLevel(AuthorizationLevel.System, keyName))
```

Comment:
This will break existing DTF and O3656 uses. 
I think we may need a corresponding update in WebJobsSdkExtensionHookProvider  to append this key. 

---
In reply to: [125786600](https://github.com/Azure/azure-webjobs-sdk-script/pull/1659#discussion_r125786600) [](ancestors = 125786600)